rpm/wrapper.h

```diff
@@ -0,0 +1,35 @@
+/*
+Patch libdnf to remove references to glibc symbols which aren't forward-compatible.
+
+This is necessary because our build system compiles libdnf with glibc-2.16, but the 
+system-probe is compiled on a system with a more recent glibc version which doesn't
+contain the __secure_getenv symbol (in glibc 2.17, the symbol was renamed secure_getenv).
+*/
+
+#include <features.h>
+
+#if defined(__GLIBC__) && defined(__GLIBC_PREREQ) && __GLIBC_PREREQ(2, 17)
+
+#define symver_wrap___secure_getenv()                                  \
+char * ____secure_getenv_glibc_2_17(char const *name);                 \
+                                                                       \
+asm(".symver ____secure_getenv_glibc_2_17, secure_getenv@GLIBC_2.17"); \
+                                                                       \
+char * __wrap___secure_getenv (char const *name) {                     \
+  return ____secure_getenv_glibc_2_17(name);                           \
+}
+
+# else
+
+// Use the function directly for older glibc / non-glibc environments
```

Comment:
I don't think you need the `else` part because the docs for `--wrap` say: 
> Any undefined reference to symbol will be resolved to "__wrap_symbol"

In the <2.17 case, the symbol exists and won't be undefined.

Reply:
🤔   I think you're right! And I suppose non-glibc environments aren't really something we need to worry about either, since this is intended for Linux systems exclusively.